addons/token-exchange/token_exchanger_agent.go

```diff
@@ -134,9 +134,10 @@ func newTokenExchangeAgentController(
 		}
 		return key
 	}
-	informer := spokeSecretInformers.Informer()
+	spokeSecretInformer := spokeSecretInformers.Informer()
+	hubSecretInformer := hubSecretInformers.Informer()
```

Comment:
This will watch all secrets. Should be updated to watch only green secrets.

Reply:
added `eventFilterFn`.  Now only reconciling if the secret has `multicluster.odf.openshift.io/secret-type: GREEN` label.